invenio_app_rdm/records_ui/searchapp.py

```diff
@@ -7,250 +7,46 @@
-            "defaultSortingOnEmptyQueryString":
-                generator_object.defaultSortingOnEmptyQueryString
-
-        }
-        config.update(kwargs)
-        return config
-
-
-#
-# Application state context processors
-#
-def sort_config(config_name):
-    """Sort configuration."""
-    return SortConfig(
-        current_app.config['RDM_SORT_OPTIONS'],
-        current_app.config[config_name].get('sort', []),
-        current_app.config[config_name].get('sort_default', None),
-        current_app.config[config_name].get('sort_default_no_query'),
-    )
-
-
-def facets_config(config_name, available_facets):
-    """Facets configuration."""
-    return FacetsConfig(
-        available_facets,
-        current_app.config[config_name].get('facets', [])
-    )
-
-
-def search_app_config(
-    config_name, available_facets, endpoint, overrides=None, **kwargs
-):
-    """Search app config."""
-    opts = dict(
-        endpoint=endpoint,
-        headers={"Accept": "application/vnd.inveniordm.v1+json"},
-        grid_view=False,
-        sort=sort_config(config_name),
-        facets=facets_config(config_name, available_facets),
-    )
-    opts.update(kwargs)
-    overrides = overrides or {}
-    return SearchAppConfig.generate(opts, **overrides)
+from invenio_search_ui.searchconfig import search_app_config
 
 
 def search_app_context():
     """Search app context processor."""
     return {
-        'search_app_config': partial(
+        'search_app_rdm_config': partial(
             search_app_config,
             'RDM_SEARCH',
             current_app.config['RDM_FACETS'],
-            '/api/records'),
-        'search_app_user_uploads_config': partial(
+            current_app.config['RDM_SORT_OPTIONS'],
+            '/api/records',
+            {"Accept": "application/vnd.inveniordm.v1+json"}
+        ),
+        'search_app_rdm_user_uploads_config': partial(
             search_app_config,
             'RDM_SEARCH_DRAFTS',
             current_app.config['RDM_FACETS'],
-            '/api/user/records'
+            current_app.config['RDM_SORT_OPTIONS'],
+            '/api/user/records',
+            {"Accept": "application/vnd.inveniordm.v1+json"},
         ),
-        'search_app_user_communities_config': partial(
+        'search_app_rdm_user_communities_config': partial(
             search_app_config,
             'RDM_SEARCH_USER_COMMUNITIES',
             current_app.config['COMMUNITIES_FACETS'],
-            '/api/user/communities', headers={"Accept": "application/json"}),
-        'search_app_user_requests_config': partial(
+            current_app.config['RDM_SORT_OPTIONS'],
+            '/api/user/communities',
+            {"Accept": "application/json"}
+        ),
+        'search_app_rdm_user_requests_config': partial(
             search_app_config,
             'RDM_SEARCH_USER_REQUESTS',
             current_app.config['REQUESTS_FACETS'],
+            current_app.config['RDM_SORT_OPTIONS'],
```

Comment:
I understand there is no sort options for the user requests and we must use the global ones, is that correct?

Reply:
Not yet (not sure if there will be), `RDM_SORT_OPTIONS` are the default ones for rdm.